### PR TITLE
feat: [CDS-84524]: setup env git details for infrastructure inputs

### PR DESCRIPTION
**Pull request by hingerabhinav**
## Describe your changes
1. update UI contract to use env info to provide branch and ref 
2. setup env git details before fetching infra inputs since env-infra are coupled entities 
3. use existing infra entity to fetch inputs, previously we were re fetching the entities 

Testing Scenarios 
1. dynamic linking of env 
2. static branch linking of env 
3. different repo of env-infra 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55453)

### CODE REVIEW

**Comment by hingerabhinav** [comment]
trigger uts, compile

**Review by adivishy1 — APPROVED** [review]
> Reviewed 7 of 8 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55453)* status: 8 file(s) to review. 1 of 2 approvals obtained. 

___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/infrastructure/services/impl/InfrastructureEntityServiceImpl.java` line 900 at r2](https://reviewable.io/reviews/harness/harness-core/55453#-NjQ_Xxb3rLA4oUTfl4q:-NjQ_Xxb3rLA4oUTfl4r:b-un0vzw) ([raw file](https://github.com/harness/harness-core/blob/2746444f6d2ee217cf060b29940a99ac8ca11e38/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/infrastructure/services/impl/InfrastructureEntityServiceImpl.java#L900)):*
> ```Java
>     Environment environment = environmentOptional.get();
>     // inline environment will always use infra from default repo
>     if (!StoreType.REMOTE.equals(environment.getStoreType())) {
> ```

why not directly check for INLINE storeType? is this same flow being invoked in the old flows as well?

**Comment by hingerabhinav** [comment]
trigger compile, uts

File: 120-ng-manager/src/main/java/io/harness/ng/core/infrastructure/resource/InfrastructureResource.java [hunk 1]
```diff
@@ -572,9 +575,10 @@ private void checkForAccessBatch(
       @BeanParam GitEntityFindInfoDTO gitEntityBasicInfo,
```
Comment (hingerabhinav) [inline 120-ng-manager/src/main/java/io/harness/ng/core/infrastructure/resource/InfrastructureResource.java:575]: add desc that this is the parent pipeline's git context

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/infrastructure/services/impl/InfrastructureEntityServiceImpl.java [hunk 2]
```diff
@@ -875,17 +883,47 @@ public List<InfrastructureYamlMetadata> createInfrastructureYamlMetadata(
...
+  @Override
+  public GitEntityInfo getGitDetailsForInfrastructure(String accountIdentifier, String orgIdentifier,
+      String projectIdentifier, String environmentRef, String environmentBranch) {
+    GitEntityInfo defaultGitContext = GitEntityInfo.builder().build();
+
+    Optional<Environment> environmentOptional =
+        environmentService.getMetadata(accountIdentifier, orgIdentifier, projectIdentifier, environmentRef, false);
+
+    if (environmentOptional.isEmpty()) {
+      return defaultGitContext;
```
Comment (tchaurasiya) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/infrastructure/services/impl/InfrastructureEntityServiceImpl.java:895]: shouldn't we throw exception??

**Review by hingerabhinav — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55453)* status: 9 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/infrastructure/services/impl/InfrastructureEntityServiceImpl.java` line 900 at r2](https://reviewable.io/reviews/harness/harness-core/55453#-NjQ_Xxb3rLA4oUTfl4q:-NjRAUwK3C_L3HEG1twZ:bitt884) ([raw file](https://github.com/harness/harness-core/blob/2746444f6d2ee217cf060b29940a99ac8ca11e38/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/infrastructure/services/impl/InfrastructureEntityServiceImpl.java#L900)):*
<details><summary><i>Previously, adivishy1 (Adithya Viswanathan) wrote…</i></summary><blockquote>

why not directly check for INLINE storeType? is this same flow being invoked in the old flows as well?
</blockquote></details>

Yes this new API will be used for inline as well. Older entities might not have store type set. 
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/infrastructure/services/impl/InfrastructureEntityServiceImpl.java` line 895 at r5](https://reviewable.io/reviews/harness/harness-core/55453#gh-1396797617:-NjRA_9F5KI9NHfiUusK:b-czsg0w) ([raw file](https://github.com/harness/harness-core/blob/7bdfd702b7fadabb29518c80def64660b3e2c607/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/infrastructure/services/impl/InfrastructureEntityServiceImpl.java#L895)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

shouldn't we throw exception??
</blockquote></details>

I don't want to add any extra validations here (on top of old flow). That will look like change in behavior. 
Since this is always called from the UI, environment ref would be correct most of the times anyway.

**Review by adivishy1 — APPROVED** [review]
> Reviewed 4 of 5 files at r3, 1 of 3 files at r5, 2 of 2 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55453)* status: 9 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by tchaurasiya — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55453)* status: 9 file(s) to review.

**Pull request merged**